src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -5957,7 +5957,7 @@ internal void TdsLogin(SqlLogin rec, TdsEnums.FeatureExtension requestedFeatures
         {
             _physicalStateObj.SetTimeoutSeconds(rec.timeout);
 
-            Debug.Assert(recoverySessionData == null || (requestedFeatures | TdsEnums.FeatureExtension.SessionRecovery) != 0, "Recovery session data without session recovery feature request");
+            Debug.Assert(recoverySessionData == null || (requestedFeatures & TdsEnums.FeatureExtension.SessionRecovery) != 0, "Recovery session data without session recovery feature request");
```

Comment:
This looks like it was supposed to be &'ing to determine whether a bit was set.  As it was written, the expression would always be true.